pkg/k8s/config_map.go

```diff
@@ -33,7 +33,9 @@ func (c *ConfigMap) EnsureExists(dataSetter configMapFiller) error {
 		return err
 	}
 	_, err = controllerutil.CreateOrUpdate(context.Background(), c.client, cm, func() error {
-		cm.Data = map[string]string{}
+		if len(cm.Data) == 0 {
```

Comment:
I think the change should be not here but in generation of the config map in Command since right now  i see that we are invoking EnsureExists method on config map which actually appends some data to the config map instead of non-incrementaly change it. The question is how this can affect other config maps? and moreover when I see that we are ensuring that a specific config map exist I understand it like the content of it will be how I see it in dataSetter. I'm not saying its a bad solution because it is working but I would suggest  not to change the interface of our k8s.ConfigMap but instead in command controller iterate over pods fill the Config map correctly for all of them and update the config map once. It will be even faster since we will reduce number of API calls.

Reply:
Done